### PR TITLE
Add "alpha phase preflight" command

### LINKED ISSUE #314: Feature Request - pre-flight command
In light of the work on supporting phases in kubeadm (#267) it would also be interesting to have the pre-flight checks be available as a phase and accessible through a specific command.

This would be useful to configuration management tools that want to first make sure everything is fine before continuing with the next phases.

### PR DESCRIPTION
**Pull request by gtirloni**
**What this PR does / why we need it:**

Adds "alpha phase preflight" command to kubeadm in order to run pre-flight checks independently of init phase.

**Which issue this PR fixes:** fixes kubernetes/kubeadm#314

/cc @luxas

### CODE REVIEW

File: cmd/kubeadm/app/preflight/checks.go
Comment (luxas) [inline cmd/kubeadm/app/preflight/checks.go]: I came to think about that you could make `RunInitMasterChecks` (and for join) call `RunRootCheckOnly`, and we'd only have one method to call. I think that would be a good improvement

File: cmd/kubeadm/app/preflight/checks.go
Comment (luxas) [inline cmd/kubeadm/app/preflight/checks.go]: I'd prefer to keep the most logging in `init.go`/`join.go`

File: cmd/kubeadm/app/cmd/phases/preflight.go
Comment (luxas) [inline cmd/kubeadm/app/cmd/phases/preflight.go]: It'd be cool if this command had two sub-commands: `master` and `node`.
Then it would be possible to run both depending on usage...

**Review by luxas — COMMENTED** [review]
Cool, thank you! That was blazing fast from issue to PR :100: 

I'd like to add the same functionality for the node preflight checks as well, sounds good?

**Comment by luxas** [comment]
@k8s-bot ok to test
/release-note-none
(the phases are very much alpha stuff still)

File: cmd/kubeadm/app/preflight/checks.go
Comment (gtirloni) [inline cmd/kubeadm/app/preflight/checks.go]: Makes sense! After all, if someone invoked `alpha phase preflight` they do know the pre-flight checks are running (as opposed to not knowing all the phases that `init` comprises.

File: cmd/kubeadm/app/preflight/checks.go
Comment (gtirloni) [inline cmd/kubeadm/app/preflight/checks.go]: Agreed. That would then eliminate the need for the new function I've created (`RunPreFlightChecks`). Just wanted to clarify the intention.

File: cmd/kubeadm/app/cmd/phases/preflight.go
Comment (gtirloni) [inline cmd/kubeadm/app/cmd/phases/preflight.go]: Oh, I forgot about `join`... Good point, I'll add those.

File: cmd/kubeadm/app/preflight/checks.go
Comment (luxas) [inline cmd/kubeadm/app/preflight/checks.go]: Can you move this above the Checker list?

Also needs to be replicated on the Node side...

File: cmd/kubeadm/app/cmd/phases/preflight.go
Comment (luxas) [inline cmd/kubeadm/app/cmd/phases/preflight.go]: Do you think there's need to be able to customize this?
I don't think we should do that in this PR though, adds more complexity than needed...

File: cmd/kubeadm/app/cmd/init.go
Comment (luxas) [inline cmd/kubeadm/app/cmd/init.go]: remove from join.go as well

**Review by luxas — COMMENTED** [review]
You need to run `hack/update-bazel.sh` for the CI to be happy as well

and `gofmt -s -w cmd/kubeadm` if you didn't already

File: cmd/kubeadm/app/preflight/checks.go
Comment (gtirloni) [inline cmd/kubeadm/app/preflight/checks.go]: Done!

File: cmd/kubeadm/app/cmd/phases/preflight.go
Comment (gtirloni) [inline cmd/kubeadm/app/cmd/phases/preflight.go]: Ideally, I think `preflight` should indeed accept all the same parameters that `init` accepts, to simulate the same environment that `init` will have when running. I can open a new issue for that.

File: cmd/kubeadm/app/cmd/init.go
Comment (gtirloni) [inline cmd/kubeadm/app/cmd/init.go]: Done!

**Review by luxas — APPROVED** [review]
Perfect, thank you!

LGTM

Only thing left to do is to squash your commits...

**Comment by gtirloni** [comment]
@luxas thanks for following me through this process. I'm learning quite a few new things. Really appreciate it.

I squashed my commits and pushed again.

**Comment by luxas** [comment]
@gtirloni My pleasure! Happy first contribution :smile:!

This won't merge until next week or so, since we're subject to code freeze, so no v1.8-targeting PRs can merge right now.

**Pull request merged**
